lib/Infer/EnumerativeSynthesis.cpp

```diff
@@ -751,70 +812,40 @@ void generateAndSortGuesses(SynthesisContext &SC,
     PruneFuncs.push_back(DataflowPruning.getPruneFunc());
   }
   auto PruneCallback = MkPruneFunc(PruneFuncs);
-  // TODO(zhengyangl): Refactor the syntactic pruning into a
-  // prune function here, between Cost and Dataflow
-  // TODO(manasij7479) : If RHS is concrete, evaluate both sides
-  // TODO(regehr?) : Solver assisted pruning (should be the last component)
 
-  getGuesses(Guesses, Cands, SC.LHS->Width,
-             LHSCost, SC.IC, nullptr, nullptr, TooExpensive, PruneCallback);
-  if (DebugLevel >= 1) {
-    DataflowPruning.printStats(llvm::errs());
-  }
+
+  std::vector<Inst *> Guesses;
+  uint64_t GuessCount = 0;
+
+  auto Generate = [&SC, &Guesses, &RHS, &EC, &GuessCount](Inst *Guess) {
+    GuessCount++;
+    Guesses.push_back(Guess);
+    if (Guesses.size() >= MaxV && !SkipSolver) {
+      sortGuesses(Guesses);
+      EC = verify(SC, RHS, Guesses);
+      Guesses.clear();
+      return RHS == nullptr; // Continue if no RHS
```

Comment:
Which RHS does this lambda capture?

Reply:
The result.

Reply:
I see, RHS is changed in verify(). It would better to rely on EC for this early termination.